### PR TITLE
Added zPCI attributes to libvirt section (jsc#SLE-4577)

### PR DESCRIPTION
**Pull request by tbazant**
### PR creator: Description

Added zPCI attributes for z-Series to the libvirt section.
Rendered output is here:
[sec-libvirt-config-pci-virsh_color_en.pdf](https://github.com/SUSE/doc-sle/files/6615086/sec-libvirt-config-pci-virsh_color_en.pdf)

### PR creator: Are there any relevant issues/feature requests?

* jsc#SLE-4577

### PR creator: Which product versions do the changes apply to?

When opening a PR, check all versions of the documentation that your PR applies to.

- SLE 15/openSUSE Leap 15.x
  - [x] SLE 15 SP4/openSUSE Leap 15.4 *(if applicable to 15 SP4 _only_ -- do not merge yet!)*
  - [x] SLE 15 SP3/openSUSE Leap 15.3 *(current `main`, no backport necessary)*
  - [x] SLE 15 SP2/openSUSE Leap 15.2
  - [x] SLE 15 SP1
  - [ ] SLE 15 GA
- SLE 12
  - [ ] SLE 12 SP5
  - [ ] SLE 12 SP4
  - [ ] SLE 12 SP3

### PR reviewer only: Have all backports been applied?

The doc team member merging your PR will take care of backporting to older documents.
When opening a PR, do *not* set the following check box.

- [ ] all necessary backports are done

### CODE REVIEW

File: xml/libvirt_configuration_virsh.xml [hunk 1]
```diff
@@ -407,7 +407,7 @@ CPU Affinity:   yy
   <title>Adding a PCI device</title>
 
   <para>
-   To assign a PCI device to &vmguest; with &virsh;, follow these steps:
+   To assign host's PCI device to &vmguest; with &virsh;, follow these steps:
```
Comment (jfehlig) [inline xml/libvirt_configuration_virsh.xml:410]: The existing wording is better IMO.

File: xml/libvirt_configuration_virsh.xml [hunk 2]
```diff
@@ -569,7 +569,32 @@ CPU Affinity:   yy
...
+     These attributes are optional and if you do not specify them, they are
+     automatically generated with non-conflicting values.
+    </para>
+    <para>
+     These attributes are implemented by using a new PCI address extension flag.
+     To include zPCI attribute in your domain specification, use the following
+     example definition:
+    </para>
+<screen>
+&lt;hostdev mode='subsystem' type='pci'>
```
Comment (jfehlig) [inline xml/libvirt_configuration_virsh.xml:589]: Need `&gt;` here and a few places below, right?

File: xml/libvirt_configuration_virsh.xml [hunk 3]
```diff
@@ -569,7 +569,32 @@ CPU Affinity:   yy
...
+ &lt;driver name='vfio'/>
+ &lt;source>
+  &lt;address domain='0x0' bus='0x03' slot='0x07' function='0x0'/&gt;
+ &lt;/source>
+ &lt;address type='pci' domain='0x0000' bus='0x00' slot='0x01' function='0x0' uid='0x0003' fid='0x00000027'/>
```
Comment (jfehlig) [inline xml/libvirt_configuration_virsh.xml:594]: I'm not familiar with s390, but this doesn't look right according to the libvirt domXML doc

https://libvirt.org/formatdomain.html#device-addresses

That text says "PCI address extensions depending on the architecture are supported. For example, PCI addresses for S390 guests will have a zpci child element, with two attributes: uid (a hex value between 0x0001 and 0xffff, inclusive), and fid (a hex value between 0x00000000 and 0xffffffff, inclusive) used by PCI devices on S390 for User-defined Identifiers and Function Identifiers."

As I understand it, zpci is a child element of  the address element. E.g.

<address type='pci' domain='0x0000' bus='0x00' slot='0x01' function='0x0'>
  <zpci uid='0x0003' fid='0x00000027'/>
</address>

And if I'm reading the schema correctly, it also defines a zpci element used by the address element

https://gitlab.com/libvirt/libvirt/-/blob/master/docs/schemas/basictypes.rng#L125
https://gitlab.com/libvirt/libvirt/-/blob/master/docs/schemas/domaincommon.rng#L6447

IMO we should also have @liayan review this change.

File: xml/libvirt_configuration_virsh.xml (hunk 1 above)
Comment (tbazant) [inline xml/libvirt_configuration_virsh.xml:410]:
```suggestion
   To assign a PCI device to &vmguest; with &virsh;, follow these steps:
```

File: xml/libvirt_configuration_virsh.xml (hunk 2 above)
Comment (tbazant) [reply xml/libvirt_configuration_virsh.xml:589]: the closing bracket is not needed in fact :-)

**Review by lproven — APPROVED** [review]
I can't comment on the tech content but the text LGTM.

**Comment by tbazant** [comment]
@jfehlig i've added an example from https://libvirt.org/pci-addresses.html#zpci-addresses

**Review by liayan — COMMENTED** [review]
It looks good, nice work here. Just one place for the beginning. It may be nice if we mention some context here. Something like: "In order to support S390, QEMU extended PCI representation by allowing to configure extra attributes. Two more attributes, uid and fid, are added to zPCI..."

**Pull request merged**